src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -716,7 +716,10 @@ private bool AnyProjectHasVulnerablePackageWarning(IReadOnlyList<RestoreSummary>
                     {
                         using SourceCacheContext sourceCacheContext = new();
                         List<SourceRepository> sourceRepositories = _sourceRepositoryProvider.GetRepositories().AsList();
+
+                        await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(token);
```

Comment:
Would it be cleaner to have this be an async method which switches to the UI thread?
That way you don't have to remember to switch back. 

Another optimization you can do is just make sure that there are packages.config projects before  switching to the UI thread for the remainder of the operation.

Reply:
> Would it be cleaner to have this be an async method which switches to the UI thread?

Depends on your definition of cleaner 🤣 
I pushed a change where I made the method async, and therefore it allocates a state machine on the heap, but it doesn't need to explicitly switch back the thread pool.

> Another optimization you can do is just make sure that there are packages.config projects before switching to the UI thread for the remainder of the operation.

I spent some time looking into this, and SolutionResotreJob already does this check before calling into this method: